worlds/factorio/__init__.py

```diff
@@ -231,7 +231,8 @@ def make_balanced_recipe(self, original: Recipe, pool: typing.Set[str], factor:
         """Generate a recipe from pool with time and cost similar to original * factor"""
         new_ingredients = {}
         # have to first sort for determinism, while filtering out non-stacking items
-        pool: typing.List[str] = sorted(pool & stacking_items)
+        pool_fluids = pool & fluids
```

Comment:
By saving fluids | stacking_items as a precomputed variable; maybe valid_ingredients, you can save some instructions per-recipe.

Reply:
Done. Once this merges, I plan to migrate this into PR #753